src/test/java/dev/jbang/cli/TestRun.java

```diff
@@ -1437,4 +1435,70 @@ void testJVMOpts() throws IOException {
 
 		assertThat(line, containsString(" --show-version "));
 	}
+
+	@Test
+	void testJavaFXViaFileDeps() throws IOException {
+		JBang jbang = new JBang();
+
+		Path fileref = examplesTestFolder.resolve("SankeyPlotTestWithDeps.java").toAbsolutePath();
+
+		// todo fix so --deps can use system properties
+		CommandLine.ParseResult pr = new CommandLine(jbang).parseArgs("run",
+				fileref.toString());
+
+		Run run = (Run) pr.subcommand().commandSpec().userObject();
+
+		RunContext ctx = run.getRunContext();
+		ctx.setMainClass("fakemain");
+		String line = run.generateCommandLine(Source.forResource(fileref.toString(), ctx), ctx);
+
+		assertThat(line, containsString(" --module-path "));
+	}
+
+	@Test
+	void testJavaFXViaCommandLineDeps() throws IOException {
+		JBang jbang = new JBang();
+
+		Path fileref = examplesTestFolder.resolve("SankeyPlotTest.java").toAbsolutePath();
+
+		// todo fix so --deps can use system properties
+		CommandLine.ParseResult pr = new CommandLine(jbang).parseArgs("run",
+				"--deps", "org.openjfx:javafx-graphics:17:mac",
+				"--deps", "org.openjfx:javafx-controls:17:mac",
+				"--deps", "eu.hansolo.fx:charts:RELEASE",
+				fileref.toString());
+
+		Run run = (Run) pr.subcommand().commandSpec().userObject();
+
+		RunContext ctx = run.getRunContext();
+		ctx.setMainClass("fakemain");
+		String line = run.generateCommandLine(Source.forResource(fileref.toString(), ctx), ctx);
+
+		assertThat(line, containsString(" --module-path "));
+	}
+
+	@Test
+	void testJavaFXMagicPropertyViaCommandline() throws IOException {
+
+		Assertions.assertThrows(ExitException.class, () -> {
+			JBang jbang = new JBang();
+
+			Path fileref = examplesTestFolder.resolve("SankeyPlotTest.java").toAbsolutePath();
+
+			// todo fix so --deps can use system properties
+			CommandLine.ParseResult pr = new CommandLine(jbang).parseArgs("run",
+					"--deps", "org.openjfx:javafx-graphics:17:${os.detected.jfxname}",
```

Comment:
this is currently not working because properties are not processed/accesssible when these --deps from command line are accessed. needs cleaning up in separate pr.